### PR TITLE
Show VAT Number on Storefront

### LINKED ISSUE #1499: Show VAT Number on Storefront (new Account Options) / M2 Support
**Description**:
Determines if a Value Added Tax number is visible on the storefront. In ScandiPWA, the VAT number is never visible on the storefront. https://docs.magento.com/user-guide/customers/account-options-new.html?itm_source=merchdocs&itm_medium=quick_search&itm_campaign=federated_search&itm_term=show%20vat%20number

**TODO**:
Implement support for the feature

### PR DESCRIPTION
**Pull request by zans-laksa**
Fixes #1499

Related PRs:
* scandipwa/quote-graphql/pull/48
* scandipwa/customer-graph-ql/pull/13
* scandipwa/store-graphql/pull/7

### CODE REVIEW

File: src/app/component/MyAccountCreateAccount/MyAccountCreateAccount.component.js [hunk 1]
```diff
@@ -16,15 +16,41 @@ import Field from 'Component/Field';
...
+        const { showTaxVatNumber } = this.props;
+        const validation = [];
+
+        if (!showTaxVatNumber) {
+            return null;
+        }
+
+        if (showTaxVatNumber === SHOW_VAT_NUMBER_REQUIRED) {
+            validation.push('notEmpty');
+        }
```
Comment (alfredsgenkins) [inline src/app/component/MyAccountCreateAccount/MyAccountCreateAccount.component.js:41]: I would rather go for vatNumberValidation coming from container.

File: src/app/component/MyAccountCustomerForm/MyAccountCustomerForm.component.js [hunk 2]
```diff
@@ -45,6 +46,27 @@ export class MyAccountCustomerForm extends FieldForm {
...
+        const { showTaxVatNumber } = this.props;
+        const validation = [];
+
+        if (!showTaxVatNumber) {
+            return {};
+        }
+
+        if (showTaxVatNumber === SHOW_VAT_NUMBER_REQUIRED) {
+            validation.push('notEmpty');
+        }
```
Comment (alfredsgenkins) [inline src/app/component/MyAccountCustomerForm/MyAccountCustomerForm.component.js:64]: I saw this logic... I do not think, however we should resolve this.

**Review by alfredsgenkins — CHANGES_REQUESTED** [review]
Front-end is pretty much alright. Can you re-do the `customer-graphql` part. Also, please resolve conflicts.

File: src/app/component/MyAccountCreateAccount/MyAccountCreateAccount.component.js (hunk 1 above)
Comment (zans-laksa) [reply src/app/component/MyAccountCreateAccount/MyAccountCreateAccount.component.js]: Done.

File: src/app/component/MyAccountCustomerForm/MyAccountCustomerForm.component.js (hunk 2 above)
Comment (zans-laksa) [reply src/app/component/MyAccountCustomerForm/MyAccountCustomerForm.component.js]: Done.

**Comment by alfredsgenkins** [comment]
Please fix ESLint and add composer dependency for https://github.com/scandipwa/store-graphql/releases/tag/1.1.2 and https://github.com/scandipwa/quote-graphql/releases/tag/2.17.2

**Pull request merged**